lib/src/Base/Optim/MultiStart.cxx

```diff
@@ -0,0 +1,164 @@
+//                                               -*- C++ -*-
+/**
+ *  @brief MultiStart optimization algorithm
+ *
+ *  Copyright 2005-2016 Airbus-EDF-IMACS-Phimeca
+ *
+ *  This library is free software: you can redistribute it and/or modify
+ *  it under the terms of the GNU Lesser General Public License as published by
+ *  the Free Software Foundation, either version 3 of the License, or
+ *  (at your option) any later version.
+ *
+ *  This library is distributed in the hope that it will be useful,
+ *  but WITHOUT ANY WARRANTY; without even the implied warranty of
+ *  MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ *  GNU Lesser General Public License for more details.
+ *
+ *  You should have received a copy of the GNU Lesser General Public
+ *  along with this library.  If not, see <http://www.gnu.org/licenses/>.
+ *
+ */
+#include "openturns/MultiStart.hxx"
+#include "openturns/PersistentObjectFactory.hxx"
+#include "openturns/Cobyla.hxx"
+#include "openturns/SpecFunc.hxx"
+#include "openturns/Box.hxx"
+
+BEGIN_NAMESPACE_OPENTURNS
+
+CLASSNAMEINIT(MultiStart);
+
+static const Factory<MultiStart> Factory_MultiStart;
+
+/* Constructor with parameters */
+MultiStart::MultiStart()
+  : OptimizationSolverImplementation()
+  , solver_(new Cobyla)
+{
+  // Nothing to do here
+}
+
+
+/* Constructor with parameters */
+MultiStart::MultiStart(const OptimizationSolver & solver,
+                       const NumericalSample & startingPoints)
+  : OptimizationSolverImplementation(solver.getProblem())
+  , solver_(solver)
+  , startingPoints_(startingPoints)
+{
+  OptimizationProblem problem(getProblem());
+  if (!problem.hasBounds())
+    throw InvalidArgumentException(HERE) << "Cannot use default multi-start without bounds when the experiment is not defined";
+
+  if (startingPoints.getDimension() != problem.getDimension())
+    throw InvalidArgumentException(HERE) << "The starting points dimension must match the problem dimension";
+}
+
+
+/* Check whether this problem can be solved by this solver.  Must be overloaded by the actual optimisation algorithm */
+void MultiStart::checkProblem(const OptimizationProblem & problem) const
+{
+  // The problem is checked in the local solver
```

Comment:
oh, yes, not in multi-start, but when you build tnc.
the idea is that the problem belongs to the internal solver (tnc) and not multistart (I changed the ownership role after the first reactions on what I introduced at the meeting)